pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -147,16 +144,9 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 			plugins = append(plugins, saAdmitter)
 
 		default:
-			configFile := server.AdmissionControlConfigFile
-			pluginConfig := options.KubernetesMasterConfig.AdmissionConfig.PluginConfig
-
-			// Check whether a config is specified for this plugin. If not, default to the
-			// global plugin config file specifiedd in the server config.
-			if cfg, hasConfig := pluginConfig[pluginName]; hasConfig {
-				configFile, err = pluginconfig.GetPluginConfig(cfg)
-				if err != nil {
-					return nil, err
-				}
+			configFile, err := oadmission.GetPluginConfigFile(options.KubernetesMasterConfig.AdmissionConfig.PluginConfig, pluginName, server.AdmissionControlConfigFile)
```

Comment:
> I prefer the old factorization, but if @csrwng prefers this I can live with it.

Since the web console asset config also needs to read this file, I thought it would be best for both to share the function to do that. Not sure if you're advocating for duplicating this code instead, or just pulling the func back into this file or something.